src/core/server/status/routes/status.ts

```diff
@@ -0,0 +1,189 @@
+}
+
+interface StatusHttpBody {
+  name: string;
+  uuid: string;
+  version: {
+    number: string;
+    build_hash: string;
+    build_number: number;
+    build_snapshot: boolean;
+  };
+  status: LegacyStatusInfo | StatusInfo;
+  metrics: {
+    /** ISO-8601 date string w/o timezone */
+    last_updated: string;
+    collection_interval_in_millis: number;
+    process: {
+      memory: {
+        heap: {
+          total_in_bytes: number;
+          used_in_bytes: number;
+          size_limit: number;
+        };
+        resident_set_size_in_bytes: number;
+      };
+      event_loop_delay: number;
+      pid: number;
+      uptime_in_millis: number;
+    };
+    os: {
+      load: Record<string, number>;
+      memory: {
+        total_in_bytes: number;
+        used_in_bytes: number;
+        free_in_bytes: number;
+      };
+      uptime_in_millis: number;
+      platform: string;
+      platformRelease: string;
+    };
+    response_times: {
+      max_in_millis: number;
+    };
+    requests: {
+      total: number;
+      disconnects: number;
+      statusCodes: Record<number, number>;
+      status_codes: Record<number, number>;
+    };
+    concurrent_connections: number;
+  };
+}
+
+const serializeStatus = (status: ServiceStatus): SerializableServiceStatus => ({
+  ...status,
+  level: status.level.toString(),
+});
+
+const serializeStatusRecord = (
+  statuses: Record<string, ServiceStatus>
+): Record<string, SerializableServiceStatus> =>
+  Object.keys(statuses).reduce((acc, serviceName) => {
+    acc[serviceName] = serializeStatus(statuses[serviceName]);
+    return acc;
+  }, {} as Record<string, SerializableServiceStatus>);
+
+export const registerStatusRoute = ({ router, config, metrics, status }: Deps) => {
+  router.get(
+    {
+      path: '/api/status',
+      options: {
+        authRequired: !config.allowAnonymous,
+      },
+      validate: {
+        query: schema.object({
+          v8format: schema.boolean({ defaultValue: false }),
+        }),
+      },
+    },
+    async (context, req, res) => {
+      const { version, buildSha, buildNum } = config.packageInfo;
+      const versionWithoutSnapshot = version.replace(SNAPSHOT_POSTFIX, '');
+
+      let statusInfo: LegacyStatusInfo | StatusInfo;
+      if (req.query?.v8format) {
+        const [overall, core, plugins] = await combineLatest([
+          status.overall$,
+          status.core$,
+          status.plugins$,
+        ])
+          .pipe(first())
+          .toPromise();
+        statusInfo = {
+          overall: serializeStatus(overall),
+          core: serializeStatusRecord((core as unknown) as Record<string, ServiceStatus>),
+          plugins: serializeStatusRecord(plugins),
+        };
+      } else {
+        statusInfo = await calculateLegacyStatus({
+          ...status,
+          versionWithoutSnapshot,
+        });
+      }
+
+      const lastMetrics = await metrics.getOpsMetrics$().pipe(first()).toPromise();
```

Comment:
this might hold the response up to `metrics.collectionInterval` (5s by default). It should only affect to the first requests during the startup process though.

Reply:
Fixed this by using a ReplaySubject in the metrics service and ensuring it's popluated before the HTTP server actually starts.